src/luarocks/cfg.lua

```diff
@@ -140,8 +143,8 @@ local platform_order = {
    -- Windows
    win32 = 10,
    mingw32 = 11,
-   windows = 12 }
-
+   windows = 12, 
+   haiku = 13 }
```

Comment:
Please put haiku after cygwin as 10 and increment the ones below it. These numbers don't need to be the same (only their order matters) but I guess it looks nicer if we keep all the Unixes and Windows together. Thanks!